mimxrt: Add ALT11 pin mode support for MIMXRT1176.

APIUM opened the pull request:
### Summary
Adds ALT11 alternate function mode support for MIMXRT1176, which enables FLEXPWM channels on GPIO_AD pins (GPIO_AD_06 through GPIO_AD_21). The RT1176 has this extra alt mode that wasn't previously exposed.

Changes: PIN_AF_MODE_ALT11 enum added to pin.h, and ALT11 column populated in MIMXRT1176_af.csv with the FLEXPWM mappings.

Only affects RT1176-based boards (MIMXRT1170_EVK and PHYBOARD_RT1170). Other MIMXRT chips use ALT0-ALT9 and are unaffected.

### Testing
Tested on MIMXRT1176 hardware with PWM output on GPIO_AD_14 (FLEXPWM3_PWM0_X).

APIUM commented:
The diff is a bit disappointing as it's word based. Would be good to match the styling for stm32 such as https://github.com/micropython/micropython/blob/master/ports/stm32/boards/stm32f722_af.csv which spaces the columns and would give a good diff for this sort of change. I didn't want to pollute this PR, and should do for all MIMXRT boards so didn't do it here.

robert-hh commented:
> Would be good to match the styling for stm32 such as https://github.com/micropython/micropython/blob/master/ports/stm32/boards/stm32f722_af.csv which spaces the columns 

Both files (STM and MIMXRT) are properly formatted csv files. If you open the csv file in file view, nice columns are shown. So it's the way github shows a diff, not the format for the csv file.

Besides that: I have built the firmware for both a 1170 MCU and a 1020 MCU. Builds fine, ADC is still working. I did not test yet PWM. For D4 (GPIO_AD_06) PWM was already available through the quad timer module. Nevertheless, you should add newly or differently supported pins to the micropython/docs/mimxrt/pinout.rst file of the board.

APIUM committed:
mimxrt: Add ALT11 pin mode support for MIMXRT1176.

This adds support for the ALT11 alternate function mode on MIMXRT1176
MCUs, enabling FLEXPWM channels on GPIO_AD pins that were previously
unavailable for PWM use.

Changes:
- Add PIN_AF_MODE_ALT11 enum to pin.h
- Add ALT11 column to MIMXRT1176_af.csv with FLEXPWM mappings
- Enables FLEXPWM1-4 on GPIO_AD_06 through GPIO_AD_21

This change only affects MIMXRT1176-based boards (MIMXRT1170_EVK and
PHYBOARD_RT1170). Other MIMXRT family chips use alternate function
modes ALT0-ALT9 and are not affected.

Tested on MIMXRT1176 hardware with PWM output on GPIO_AD_14
(FLEXPWM3_PWM0_X).

Signed-off-by: Algy Tynan <algy@tynan.io>

dpgeorge reviewed (approved):
This looks fine to me.  Thanks for also updating the docs.
